cypress/support/step_definitions/organisation/details/details.ts

```diff
@@ -0,0 +1,44 @@
+import { Given, Then, When } from "cypress-cucumber-preprocessor/steps";
+import { getOrganisationId } from "../../../utils";
+import { nanoid } from "nanoid";
+
+Given("an organization exists", () => {
+  const organisationId = getOrganisationId();
+  cy.createOrganisation(organisationId);
+});
+
+Given("a source organisation exists", () => {
+  const organisation = getOrganisationId();
+  cy.createMyOrgSource(organisation);
+});
+
+Given("the user is on the organisation details page", () =>
+  cy.visit(`/orgs/${getOrganisationId()}`)
+);
+
+Then("the user should see details of the organisation", () => {
+  cy.findByText("Details").should("be.visible");
+  cy
+    .getOrganisation(getOrganisationId())
+    .its("name")
+    .should("eq", "Test Organisation");
+  cy.findByText("Sources", { selector: "legend" }).should("be.visible");
+  cy.findByText("Members", { selector: "legend" }).should("be.visible");
+  cy.findByText("Dictionaries", { selector: "legend" }).should("be.visible");
+});
+
+Then("the user should see the organisation source", () => {
+  const organisation = getOrganisationId();
+  cy.createMyOrgSource(organisation);
+  cy.getOrgSources(organisation);
+  cy.findByText("My Org Source", { selector: "li" }).should("be.visible");
+  cy.get('ul').contains('li', 'My Org Source');
+});
+
+When("the user clicks on the source", () =>
+  cy.get('ul li a').click()
```

Comment:
This might work but it does so more or less by accident. `ul li a` is the selector for any anchor element that is a child of an li element which is a child of a ui element. It will select one result for each source and / or collection that the organisation has.

A better strategy would be to find the source link by a known value. This is why we tend to use `cy.findByText()` because it ensures that we're looking for the element the way the user would (i.e., by it's displayed text value).